api/src/Util/JsonMergePatch.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Util;
+
+class JsonMergePatch {
+    /**
+     * implements JSON merge patch (https://datatracker.ietf.org/doc/html/rfc7386)
+     * partially copied by discussion under https://www.php.net/manual/de/function.array-merge-recursive.php.
+     */
+    public static function mergePatch(array $target, array $patch) {
```

Comment:
Ich würde mir bei selbst implementierten Algorithmen wie diesem hier ausführliche Unit Tests für alle denkbaren happy paths, sad paths und edge cases wünschen.

Reply:
Fully agree and mentally already on my list 😄 
Potentially in separate PR, let's discuss

Reply:
Will be fixed in #2944